File: app/src/main/java/com/hypnoweb/hypnowebapp/splashscreen/SplashViewModel.java
package com.hypnoweb.hypnowebapp.splashscreen;

import android.app.Application;
import android.arch.lifecycle.AndroidViewModel;
import android.databinding.ObservableBoolean;
import android.widget.Toast;

import com.hypnoweb.hypnowebapp.catalog.repo.CatalogRepository;

import javax.inject.Inject;

import io.reactivex.annotations.NonNull;


public class SplashViewModel extends AndroidViewModel {

    public ObservableBoolean loading = new ObservableBoolean();
    private CatalogRepository catalogRepository;


    @Inject
    public SplashViewModel(CatalogRepository catalogRepository, @NonNull Application application) {
        super(application);
        this.catalogRepository = catalogRepository;
        loading.set(true);
        Toast.makeText(application, "Created View", Toast.LENGTH_SHORT).show();
    }



    /*public void addSubscriptions(CompositeDisposable compositeDisposable) {
        catalogRepository.getCatalog()
                .subscribeOn(Schedulers.io())
                .subscribe(catalog -> loading.set(false),
                        throwable -> Timber.e(throwable));

    }*/
}

File: app/src/main/java/com/hypnoweb/hypnowebapp/splashscreen/SplashViewModel.kt
package com.hypnoweb.hypnowebapp.splashscreen

import android.arch.lifecycle.ViewModel
import android.databinding.ObservableBoolean
import com.hypnoweb.hypnowebapp.catalog.repo.CatalogRepository
import com.hypnoweb.hypnowebapp.splashscreen.SplashScreenNavigator
import io.reactivex.disposables.CompositeDisposable
import io.reactivex.schedulers.Schedulers
import timber.log.Timber
import javax.inject.Inject

class SplashViewModel @Inject constructor(private val catalogRepository: CatalogRepository,
                                          private val splashScreenNavigator: SplashScreenNavigator) : ViewModel() {

    val bag: CompositeDisposable
    var loading = ObservableBoolean()

    init {
        bag = CompositeDisposable()
        loading.set(true)
    }

    fun addSubscriptions() {
        bag.add(catalogRepository.catalog
                .subscribeOn(Schedulers.io())
                .subscribe(
                        { catalog ->
                            loading.set(false)
                            splashScreenNavigator.goToCatalog()
                        },
                        { throwable -> Timber.e(throwable) }
                )
        )
    }

    override fun onCleared() {
        super.onCleared()
        bag.dispose()
    }
}
